taichi/transforms/offload.cpp

```diff
@@ -331,9 +340,11 @@ class FixCrossOffloadReferences : public BasicStmtVisitor {
  private:
   FixCrossOffloadReferences(
       const StmtToOffsetMap &local_to_global_offset,
-      std::unordered_map<Stmt *, Stmt *> stmt_to_offloaded)
+      const std::unordered_map<Stmt *, Stmt *> &stmt_to_offloaded,
```

Comment:
Any reason why `const &` here?

Reply:
Good point. If you look at the entire call stack, it was like:

```
offload()
\- FixCrossOffloadReferences::run()
    \- FixCrossOffloadReferences::FixCrossOffloadReferences()
```

Before this PR, only the last call used `move`, while the first two didn't (likely because we forgot to), so we still made two unnecessary copies. While move is a modern feature, IMO the easiest way to avoid copy remains using `const &`. (`move` also means transferring ownership, which is really useful when data cannot be copy constructed, e.g. `std::unique_ptr`). Does this make sense?